BraveRewardsUI/Tipping/TippingViewController.swift

```diff
@@ -72,9 +148,31 @@ class TippingViewController: UIViewController, UIViewControllerTransitioningDele
   }
   
   @objc private func tappedSendTip() {
-    tippingView.setTippingConfirmationVisible(true, animated: true)
-    DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { [weak self] in
-      self?.dismiss(animated: true)
+    if let selectedIndex = self.tippingView.optionSelectionView.selectedOptionIndex {
+      let amount = self.tippingView.optionSelectionView.options[selectedIndex].value.doubleValue
+      
+      state.ledger.listRecurringTips { [weak self] info in
+        guard let self = self else { return }
+        
+        // If the user has recurring tips, remove the monthly tip
+        if info.firstIndex(where: { $0.id == self.publisherInfo.id }) != nil {
+          self.state.ledger.removeRecurringTip(publisherId: self.publisherInfo.id)
+        }
+        
+        // Add recurring tips if there is none..
+        if self.tippingView.optionSelectionView.isMonthly {
+          self.state.ledger.addRecurringTip(publisherId: self.publisherInfo.id, amount: amount)
+        }
+        
+        // TODO: Figure out -- Not sure why amount is of type integer..
+        // Also not sure user's currency
+        self.state.ledger.tipPublisherDirectly(self.publisherInfo, amount: Int32(amount), currency: "USD")
+        self.tippingView.setTippingConfirmationVisible(true, animated: true)
```

Comment:
The tipping confirmations screen has the correct favicon now? And I believe it can have different messages appear based on whether or not this was a monthly tip or not

Reply:
Fixed.